src/components/notifications/Message.js

```diff
@@ -65,7 +67,25 @@ function AnalysisLink(props) {
         if (analysisId) {
             const [href, as] = getAnalysisDetailsLinkRefs(analysisId);
 
-            return <MessageLink message={message} href={href} as={as} />;
+            const allowRating =
+                isJobStatusChange &&
+                notification.payload?.system_id === systemIds.de &&
+                (notification.payload?.status === analysisStatus.COMPLETED ||
+                    notification.payload?.status === analysisStatus.FAILED ||
+                    notification.payload?.status === analysisStatus.CANCELED);
+            return (
+                <>
+                    <MessageLink message={message} href={href} as={as} />
+                    <br />
+                    {allowRating && (
+                        <RatingWidget
+                            appId={notification.payload?.app_id}
+                            appName={notification?.payload.app_name}
```

Comment:
Other places in this code are checking for `notification.payload?.`, but here it's `notification?.payload.`

Maybe these could be refactored into a `payload` var that replaces all of them?
```javascript
const payload = notification?.payload;
```
Then `payload?.app_name` etc.